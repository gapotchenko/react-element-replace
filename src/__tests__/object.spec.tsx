import * as React from 'react';
import { Replacer } from "..";
import { create } from 'react-test-renderer';

const { useState } = React;

function TestComponentWithObject(props: { label?: string, value: string, children: any }) {
    const [label, setLabel] = useState<any>("a string");
    return (
        <div>
            <div>Test</div>
            <span>A span</span>
            <div>{{ comment: props.value }}</div>
            <div>{props.children}</div>
            <div>{label}</div>
            <button onClick={() => setLabel({ test: "an object" })}>Testing</button>
        </div>
    );
}

test("object to string replace", () => {
    const component = create(
        <Replacer
            matchLiteral={i => typeof i === "object"}
            replace={i => <>{JSON.stringify(i)}</>}>
            <TestComponentWithObject value="text">
                <TestComponentWithObject value="text">
                    {{ test: 123 }}
                </TestComponentWithObject>
            </TestComponentWithObject>
        </Replacer>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    

});
