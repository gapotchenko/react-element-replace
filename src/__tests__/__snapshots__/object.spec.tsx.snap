// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object to string replace 1`] = `
<div>
  <div>
    Test
  </div>
  <span>
    A span
  </span>
  <div>
    {"comment":"text"}
  </div>
  <div>
    <div>
      <div>
        Test
      </div>
      <span>
        A span
      </span>
      <div>
        {"comment":"text"}
      </div>
      <div>
        {"test":123}
      </div>
      <div>
        a string
      </div>
      <button
        onClick={[Function]}
      >
        Testing
      </button>
    </div>
  </div>
  <div>
    a string
  </div>
  <button
    onClick={[Function]}
  >
    Testing
  </button>
</div>
`;

exports[`object to string replace 2`] = `
<div>
  <div>
    Test
  </div>
  <span>
    A span
  </span>
  <div>
    {"comment":"text"}
  </div>
  <div>
    <div>
      <div>
        Test
      </div>
      <span>
        A span
      </span>
      <div>
        {"comment":"text"}
      </div>
      <div>
        {"test":123}
      </div>
      <div>
        {"test":"an object"}
      </div>
      <button
        onClick={[Function]}
      >
        Testing
      </button>
    </div>
  </div>
  <div>
    {"test":"an object"}
  </div>
  <button
    onClick={[Function]}
  >
    Testing
  </button>
</div>
`;
